Add postgres query for packages response

coderbydesign opened the pull request:
Sample response from `/content/packages`
```json
{
  "data": [
    {
      "advisory_id": 1,
      "description_hash": "foo",
      "evra": "1.2.321-54.x86_64",
      "name_id": 101,
      "summary_hash": "bar",
      "synced": true
    },
    {
      "advisory_id": 1,
      "description_hash": "baz",
      "evra": "72.3.432.3-2344.x86_64",
      "name_id": 102,
      "summary_hash": "fizz",
      "synced": false
    }
  ]
}
```
**UPDATED IN [8fa9b38](https://github.com/merlante/inventory_access_poc/pull/8/commits/8fa9b38bc0fe01684a91d7885cc8b33add08e13b):**
```json
{
  "data": [
    {
      "name": "kernel",
      "summary": "The Linux kernel",
      "systems_applicable": 43,
      "systems_installable": 45,
      "systems_installed": 56
    }
  ]
}
```

coderbydesign committed:
Add initial postgres query/json response

- adds `PackagePayload` struct for a packages list
- adds `PreFilterServer#QueryPackages` which returns a `PackagePayload` after
  querying the database for package records
- returns the results of `QueryPackages` as JSON from `/content/packages`

coderbydesign committed:
Add remaining fields and account for nullable fields

coderbydesign committed:
Adjust payload and naming

coderbydesign committed:
Simplify nullable struct pointer

abaiken commented:
I pulled this branch and tested the endpoint: 
```
{
  "data": [
    {
      "name_id": 2,
      "evra": "1.1-1.el8.x86_64",
      "description_hash": "MA==",
      "summary_hash": "MA==",
      "advisory_id": 2,
      "synced": false
    },
    {
      "name_id": 3,
      "evra": "2.2-1.el8.x86_64",
      "description_hash": "MA==",
      "summary_hash": "MA==",
      "advisory_id": 3,
      "synced": false
    },
    {
      "name_id": 4,
      "evra": "3.3-1.el8.x86_64",
      "description_hash": "MA==",
      "summary_hash": "MA==",
      "advisory_id": 4,
      "synced": false
    },
```

coderbydesign committed:
Update query/structs

Updated based on this `packagesQuery` [1]. This requires data in `rh_account` and
`package_account_data`. A simple bit of seed data will get data to return:

```
INSERT INTO rh_account (name, org_id, valid_package_cache, valid_advisory_cache)
VALUES('Acme',
       100001,
       TRUE,
       TRUE);

INSERT INTO package_account_data (package_name_id, rh_account_id, systems_installed, systems_installable, systems_applicable)
VALUES(101,
       100,
       56,
       45,
       43);
```

[1] https://github.com/RedHatInsights/patchman-engine/blob/2864f268895cb8f3a2ed59e42b56ced22787c3da/manager/controllers/packages.go#L87
